File: dotnet/Multiplayer/Rooms/RoomCreateRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms
{
    public class RoomCreateRequest<TPlayerData, TRules>
    where TPlayerData : class, new() where TRules : class, new()
    {
        [JsonInclude]
        private string Room_name { get; set; } = string.Empty;
        [JsonInclude]
        private string? Password { get; set; }
        [JsonInclude]
        private int Max_players { get; set; }
        [JsonInclude]
        private bool Host_switch { get; set; }
        [JsonInclude]
        private bool Realtime { get; set; }
        [JsonInclude]
        private TPlayerData? Player_data { get; set; }
        [JsonInclude]
        private TRules? Rules { get; set; }



        public RoomCreateRequest(string room_name, int max_players, string? password = null,
            bool hostSwitch = false, bool realtime = false, TPlayerData? playerData = null, TRules? rules = null)
        {
            Room_name = room_name;
            Password = password;
            Max_players = max_players;
            Host_switch = hostSwitch;
            Realtime = realtime;
            Player_data = playerData;
            Rules = rules;
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/RoomCreateRequest.java
package com.michitai.multiplayer.rooms;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Request for creating a new game room.
 */
public class RoomCreateRequest {
    @JsonProperty("room_name")
    private String roomName;

    @JsonProperty("password")
    private String password;

    @JsonProperty("max_players")
    private int maxPlayers;

    @JsonProperty("host_switch")
    private boolean hostSwitch;

    @JsonProperty("realtime")
    private boolean realtime;

    @JsonProperty("player_data")
    private String playerData;

    @JsonProperty("rules")
    private String rules;

    public RoomCreateRequest(String roomName, String password, int maxPlayers, boolean hostSwitch, boolean realtime, String playerData, String rules) {
        this.roomName = roomName;
        this.password = password;
        this.maxPlayers = maxPlayers;
        this.hostSwitch = hostSwitch;
        this.realtime = realtime;
        this.playerData = playerData;
        this.rules = rules;
    }

    public String getRoomName() {
        return roomName;
    }

    public void setRoomName(String roomName) {
        this.roomName = roomName;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public int getMaxPlayers() {
        return maxPlayers;
    }

    public void setMaxPlayers(int maxPlayers) {
        this.maxPlayers = maxPlayers;
    }

    public boolean isHostSwitch() {
        return hostSwitch;
    }

    public void setHostSwitch(boolean hostSwitch) {
        this.hostSwitch = hostSwitch;
    }

    public boolean isRealtime() {
        return realtime;
    }

    public void setRealtime(boolean realtime) {
        this.realtime = realtime;
    }

    public String getPlayerData() {
        return playerData;
    }

    public void setPlayerData(String playerData) {
        this.playerData = playerData;
    }

    public String getRules() {
        return rules;
    }

    public void setRules(String rules) {
        this.rules = rules;
    }
}
